### DISCUSSION TITLE
How to properly deal with `readonly` values

### DISCUSSION THREAD

**Discussion by freemanzMrojo** [wevm/viem#2671] [Question]
Good afternoon,

Hope you are doing well. Maybe this question is duplicated, so sorry in advance. I was not able to find the same discussion :(

I have noticed that multiple methods are dealing `readonly` values (i.e., `parseAbiParameters`, `decodeAbiParameters`...) so sometimes when operating with mutable variables I end up having code that I would like to avoid, like:

```
// here decodedParams would be readonly, + I need to make a copy of the mutableAbiParameter
const decodedParams = decodeAbiParameters([...mutableAbiParameter], data); 

// here parsedAbiParameters would also be readonly
const parsedAbiParameters = parseAbiParameters(mutableApiParameterAsString); 
```

So I was wondering if there is a clean way to proceed in these cases that you recommend, other than the obvious solution of turning my params into `readonly` params. thank you in advance!

**Comment by tmm** [comment]
Can you share a [TypeScript Playground](https://www.typescriptlang.org/play) with what you are running into?

**Reply by freemanzMrojo** [reply]
Sorry, I was on hols, will provide a sample soon 👍
